### PR TITLE
Missing Content-Type: create critical sibling of 920340 in PL2

### PR DESCRIPTION
**Pull request by lifeforms**
As per discussion during the June IRC meet, we leave 920340 as a NOTICE (since it is technically allowed by the spec) but create a CRITICAL copy of the rule in paranoia level 2.

### CODE REVIEW

**Comment by dune73** [comment]
This works as advertised.

Two minor things:

### Typo

Could you capitalize "PL2: block" please?

### Message

The two rules have identical messages. This is a bit odd:

```
920340 Request Containing Content, but Missing Content-Type header
920341 Request Containing Content, but Missing Content-Type header
```

I'm not sure how to bring a clearer message across.

**Comment by fgsch** [comment]
I'm a bit confused. While the severity is different both rules are exactly the same. e.g.

> PL2: block on Missing Content-Type Header with Request Body

The part about blocking is true for both rules.

Am I missing something?

**Comment by dune73** [comment]
You got this right. Here is why:

920340 only brings 2 points. But it's a major evasion as you can work around URLENCODED bodyprocessor this way. Now @spartantri thinks mobile clients omit the header a lot, while @lifeforms says he hardly sees any FPs. As long as we do not know for sure we do not want to raise the severity of 920340, however, we want to make sure to block this evasion at PL2 by default. Hence the duplication via 920341.

I think it is odd, to bring a stricter twin of a rule, but here, I personally think it is appropriate.

**Comment by fgsch** [comment]
@dune73 Not sure if my point was understood so let me rephrase:  based on the comment on 920341, shouldn't 920340 be `pass`ing?

Re mobile clients, maybe look at httparchive to confirm?

**Comment by dune73** [comment]
Got you now, thanks.

@lifeforms, what do you think?

I am not really into this blocking / passing in CRS as I always run in scoring mode.

**Comment by fgsch** [comment]
FTR, all the rules in NOTICE are pass, with the exception of this one.
If we're adding the same rule with block, I think it makes sense to change 920340 to pass as well.

**Comment by dune73** [comment]
ACK

**Comment by lifeforms** [comment]
Action on 920340 set to pass. Comments and message of 920341 updated.

**Comment by dune73** [comment]
Thank you @lifeforms. Ready to merge AFAICS.
